subsys/fs/littlefs_fs.c

```diff
@@ -184,6 +195,32 @@ static int lfs_api_erase(const struct lfs_config *c, lfs_block_t block)
 	return errno_to_lfs(rc);
 }
 
+#if IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)
```

Comment:
Looking at all these ifdefery, would it make sense to have littlefs_fs_blk.c, littlefs_mtd.c, littlefs_common.c?
Simultaneous use of both blk and mtd backend at the same time is covered by this changes, right?

Reply:
Yes, the DISK flag is used with mount so that system could recognize at which storage support to use.
I'd rather postpone the division, lets now get the blk support and we can then work on making the littlefs shim pretty.

Reply:
> Looking at all these ifdefery, would it make sense to have littlefs_fs_blk.c, littlefs_mtd.c, littlefs_common.c? 

As fair as I can tell - 6 `#if IS_ENABLED()` have been added.

> Simultaneous use of both blk and mtd backend at the same time is covered by this changes, right?

Yes, the goal is to use `littlefs` on both mediums - `sd card` and (as it was supported by default) `flash` memory.
It shall be also possible to use the `littlefs` only on flash. However, it will not be possible to use `littlefs` only on SDMMC (and disable the flash part)

Reply:
> I'd rather postpone the division, lets now get the blk support and we can then work on making the littlefs shim pretty.

I do not see any reason to postpone it instead of having clear design with two backends.



> However, it will not be possible to use `littlefs` only on SDMMC (and disable the flash part)

Well that is bad. I suggest to go a step back, move or add mtd specific code to littlefs_mtd.c and finally add block device backend. I request that as change and block this PR for now.